### PR TITLE
Initial version of solr support functions

### PR DESCRIPTION
**Pull request by zac-poe**
Provides functions for:
- value source for generically mapping any content type with a repeating group keys to values
- value source for mapping a taxonomy keys to values

(Use of these functions in solr queries also requires solrconfig.xml valuesource configuration per value source)

### CODE REVIEW

**Comment by zac-poe** [comment]
The steps below define how to configure and test this code.

Setup:
1. Define a Crafter Taxonomy
1.1. Add one or more key/value pairs to this taxonomy
1. Define a Crafter Content type
2.1. Add a site component data source to the taxonomy created above
2.2. Add a metadata dropdown field which points to this datasource
2.3. Create one or more instances of this content type
1. Build and deploy this jar to {crafter-home}/bin/solr/server/solr/configsets/crafter_configs/lib
1. Add configuration to these new value sources
4.1. Add the dependent solr core jars, if not already present: `<lib dir="${solr.install.dir:../../../..}/dist/" regex="solr-core-\d.*\.jar" />`
4.2. Add the function reference and name (as desired) for the taxonomy value source: `<valueSourceParser name="taxonomyValue" class="org.crafter.solr.contentType.taxonomy.TaxonomyValueSourceParser" />`
4.3. Add the function reference and name (as desired) for the generic content type value source: `<valueSourceParser name="contentTypeValue" class="org.crafter.solr.contentType.RepeatingGroupContentTypeValueSourceParser" />`

_The examples below assume:_
- A taxonomy called "Countries"
- A content type called "Asset"
- A metadata field on "Asset" called "country_s"

**taxonomyValue parameters:**
1. internal name of Taxonomy
1. key field (or literal) to map to taxonomy value

**contentTypeValue parameters:**
1. Crafter content type
1. field name of unique ID for this content type
1. value of unique ID for this content type
1. key field name for repeating group
1. value field name for repeating group
1. key field (or literal) to map to taxonomy value

Querying Examples:
-  Execute a query for `content-type: "/component/asset"` and observe country_s shows up as a key string
- Modify the fl for the query above to: `country_s,taxonomyValue("Countries",country_s)` and observe the result documents containing both the key field from the document and the mapped taxonomy value
- Remove the fl query and add `taxonomyValue("Countries",country_s)` to the query sort and observe that the resulting documents are sorted by the taxonomy Country _value_ instead of the key
- The same can be repeated with _any_ Crafter content type containing a repeating group through use of the RepeatingGroupContentTypeValueSourceParser. Use `contentTypeValue("/component/taxonomy","internal-name","Countries","items.item.key","items.item.value",country_s)` to get the same effect against a taxonomy. Alternatively, create a new Crafter content type as a data source for your Asset and use this function appropriately with that new content type.

File: src/main/java/org/crafter/solr/SimpleIndexSearcher.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ */
+public class SimpleIndexSearcher {
+	private final IndexSearcher searcher;
+	
+	public SimpleIndexSearcher(IndexReaderContext reader) {
+		this.searcher = new IndexSearcher(getParnetContext(reader));
+	}
+
+	private IndexReaderContext getParnetContext(IndexReaderContext reader) {
```
Comment (jdrossl) [inline src/main/java/org/crafter/solr/SimpleIndexSearcher.java]: Small typo, `Parnet`

**Comment by jdrossl** [comment]
@zac-poe maybe I think it would be good to include a readme file with the build & testing instructions from your last comment

**Comment by zac-poe** [comment]
@jdrossl, great point. I added a README.

**Comment by zac-poe** [comment]
Just noticed the packages are org.crafter instead of org.craftercms - should this be updated?
Additionally, please let me know if any general renaming is preferred (root package, classes, etc).

**Comment by jdrossl** [comment]
Yes, for consistency the package should be `org.craftercms.solr`.
And please add the following notice at the beginning of all java files:
```
/*
 * Copyright (C) 2007-2018 Crafter Software Corporation. All Rights Reserved.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
```

**Pull request merged**